### DISCUSSION TITLE
Check if key exists?

### DISCUSSION THREAD

**Discussion by raj-altrio** [vercel/swr#498] [General] [answered]
Is there any way to check if a key exists in the swr cache?

I do this little trick sometimes where I preload the next key I am going to fetch with the current data to avoid any change in the UI (fallback to Suspense, etc.) while the data is loading. The drawback of this of course is that I lose the benefit of the cache, which is a pretty major drawback.

Ideally, I would check if the next key exists in the cache and, if so, do nothing. But I can't find a way to check the existence of a key.

Any help would be much appreciated.

**Comment by huozhi** [comment, marked as answer]
@raj-altrio hope `cache.has(key)` api is the thing you're looking for

```js
import {cache} from 'swr'

cache.set(key, value)
cache.has(key) // return true
```

**Reply by raj-altrio** [reply]
That worked. Thank you. Didn’t realize that was exported.

**Comment by AliReza99** [comment]
How can we do it using swr v2 ?
